lib/krane/deploy_task.rb

```diff
@@ -277,16 +277,42 @@ def validate_configuration(prune:)
     end
     measure_method(:validate_configuration)
 
+    def partition_dry_run_resources(resources)
+      individuals = []
+      mutating_webhooks = cluster_resource_discoverer.fetch_mutating_webhook_configurations
+      mutating_webhooks.each do |spec|
+        spec.dig('webhooks').each do |webhook|
```

Comment:
This makes me think we want a `MutatingWebhookConfigurations` class instead of putting all of the logic in this function.

Reply:
I made a few attempts at this. Modelling it as a `KubernetesResource` didn't seem right, but neither did modelling it in the top-level namespace. If I had to make a choice, I'm tempted to do the latter.

Reply:
Maybe its time for an `Internal` namespace for objects that are K8s objects but aren't designed for having deploy logic?

Reply:
I'm loath to make the separation... ultimately this is still just a kubernetes object. For now, I've made it a subclass of `KubernetesResource` and added the appropriate overrides should one of these ever be deployed.